### PR TITLE
Move saneyaml.py and its tests from licensedcode to commoncode #831

### PR DESCRIPTION
**Pull request by SaravananOffl**
Moved Saneyaml.py  #831 

   * saneyaml.py and its tests are not specific to licensed code.

Signed-off-by: Saravanan G <sgoffl1603@gmail.com>

### CODE REVIEW

**Comment by pombredanne** [comment]
Thank you! 
Please fix the code references in `licensecode.saneyaml` to `commoncode.saneyaml`

You can grep the code or see this by running the tests locally with `py.test -vvs` or check the CI builds here that are failing to see what needs to be changed in terms of imports.

**Comment by pombredanne** [comment]
See https://ci.appveyor.com/project/nexB/scancode-toolkit/build/1413/job/7pswer9h7kuhksdc#L102

**Comment by SaravananOffl** [comment]
It's showing error in models.py (Under Licensed code) only. So, Is it enough to change code references in model.py alone? or do I've to look each and every file?

**Comment by pombredanne** [comment]
You did not move the test data from https://github.com/nexB/scancode-toolkit/tree/develop/tests/licensedcode/data/saneyaml to `tests/commoncode/data` hence your test failure

File: src/licensedcode/models.py [hunk 1]
```diff
@@ -37,21 +37,25 @@
 from os.path import exists
 from os.path import join
 
+
```
Comment (pombredanne) [inline src/licensedcode/models.py]: Why this extra line?

File: tests/licensedcode/test_detection_datadriven.py [hunk 2]
```diff
@@ -51,6 +51,7 @@
 Data-driven tests using expectations stored in YAML files.
 """
 
+
```
Comment (pombredanne) [inline tests/licensedcode/test_detection_datadriven.py]: Why this extra line?

File: tests/licensedcode/test_detection_datadriven.py [hunk 3]
```diff
@@ -70,6 +71,7 @@ class LicenseTest(object):
     If the list of licenses is empty, then this test should not detect any
     license in the test file.
     """
+
```
Comment (pombredanne) [inline tests/licensedcode/test_detection_datadriven.py]: Why this extra line?

File: src/licensedcode/models.py [hunk 4]
```diff
@@ -37,21 +37,25 @@
...
 from licensedcode import licenses_data_dir
 from licensedcode import rules_data_dir
-from licensedcode import saneyaml
 from licensedcode.tokenize import rule_tokenizer
 from licensedcode.tokenize import query_tokenizer
 from commoncode import fileutils
 
 
+
+
```
Comment (pombredanne) [inline src/licensedcode/models.py]: Why these extra lines?

**Comment by SaravananOffl** [comment]
I'm sorry for that. I'll edit and submit my final version.

**Comment by SaravananOffl** [comment]
@pombredanne  Sir, Did you check?.Is there anything more to be modified?

File: clear [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+ [m
+[31m-[m
+ """[m
+ Data-driven tests using expectations stored in YAML files.[m
+ """[m
+ [m
+[31m-[m
+ class LicenseTest(object):[m
+     """[m
+     A license detection test is used to verify that license detection works[m
```
Comment (pombredanne) [inline clear]: What is this file?

File: src/licensedcode/models.py [hunk 6]
```diff
@@ -40,18 +40,18 @@
 from commoncode.fileutils import file_base_name
 from commoncode.fileutils import file_name
 from commoncode.fileutils import file_iter
+from commoncode import saneyaml
+
```
Comment (pombredanne) [inline src/licensedcode/models.py]: This empty line is not needed

File: src/licensedcode/models.py [hunk 7]
```diff
@@ -40,18 +40,18 @@
...
 from licensedcode import MIN_MATCH_LENGTH
 from licensedcode import MIN_MATCH_HIGH_LENGTH
 from licensedcode import licenses_data_dir
 from licensedcode import rules_data_dir
-from licensedcode import saneyaml
 from licensedcode.tokenize import rule_tokenizer
 from licensedcode.tokenize import query_tokenizer
 from commoncode import fileutils
 
-
```
Comment (pombredanne) [inline src/licensedcode/models.py]: Why removing this line?

File: src/licensedcode/models.py [hunk 8]
```diff
@@ -60,7 +60,6 @@
 # Set to True to print detailed representations of objects when tracing
 TRACE_REPR = False
 
-
```
Comment (pombredanne) [inline src/licensedcode/models.py]: Why removing this line?

File: tests/licensedcode/test_detection_datadriven.py [hunk 9]
```diff
@@ -35,18 +35,15 @@
 
 from commoncode import fileutils
 from commoncode import text
-
-from licensedcode import saneyaml
+from commoncode import saneyaml
 
 from license_test_utils import make_license_test_function
 
-
```
Comment (pombredanne) [inline tests/licensedcode/test_detection_datadriven.py]: Why removing this line?

File: tests/licensedcode/test_detection_datadriven.py [hunk 10]
```diff
@@ -35,18 +35,15 @@
...
 TEST_DATA_DIR = os.path.join(os.path.dirname(__file__), 'data/licenses')
 
 # set to True to print matched texts on test failure.
 TRACE_TEXTS = True
 
-
```
Comment (pombredanne) [inline tests/licensedcode/test_detection_datadriven.py]: Why removing this line?

**Comment by SaravananOffl** [comment]
@pombredanne I've rectified all the mistakes you've mentioned earlier.I'm sorry for those silly mistakes!

**Comment by pombredanne** [comment]
LGTM ! Thanks++

**Comment by SaravananOffl** [comment]
@pombredanne So will this PR gets merged?

**Comment by pombredanne** [comment]
I squashed your commits and forced push to your PR's branch.

**Comment by pombredanne** [comment]
I added your Signed-of-by to the commit BTW, your commits were missing it

**Comment by SaravananOffl** [comment]
Awesome! .Thanks for guiding me!!!!

**Pull request merged**